Review  and fix RPC call messages

luisantoniocrag opened the pull request:
close #328 

This PR reviews and edits ALL rpc call help messages.

luisantoniocrag committed:
Fix RPC misc call help messages

luisantoniocrag committed:
Fix RPC blockchain call help messages

luisantoniocrag committed:
Fix RPC mining call help messages

luisantoniocrag committed:
Fix RPC net call help messages

luisantoniocrag committed:
Fix RPC rawtransaction call help messages

luisantoniocrag committed:
Fix RPC rpcdisclosure call help messages

luisantoniocrag committed:
Fix RPC server call help messages

luisantoniocrag committed:
Fix RPC rpcdump call help messages

luisantoniocrag committed:
Fix RPC rpcwallet call help messages

luisantoniocrag committed:
Fix misc rpc help messages

luisantoniocrag committed:
Fix rpcwallet

luisantoniocrag committed:
Merge branch 'development' into review_rpc_call_msgs

sonzap006 reviewed (commented):
n

sonzap006 reviewed (commented):
Good

cronicc commented on src/rpc/blockchain.cpp:
```diff
@@ -193,9 +195,11 @@ UniValue getbestblockhash(const UniValue& params, bool fHelp)
         throw runtime_error(
             "getbestblockhash\n"
-            "\nReturns the hash of the best (tip) block in the longest block chain.\n"
+            "\nReturns the hash of the best (most recent) block in the longest block chain.\n"
```
```suggestion
            "\nReturns the hash of the best (most recent) block in the active block chain.\n"
```
Best to word it like this, as with our penalty system for delayed block submission it's possible that the longest chain is not the active/best chain.

cronicc commented on src/rpc/blockchain.cpp:
```diff
@@ -400,45 +416,60 @@ UniValue getblock(const UniValue& params, bool fHelp)
             "{\n"
-            "  ...,                     Same output as verbosity = 2.\n"
-            "  \"tx\" : [               (array of Objects) The transactions in the format of the getrawtransaction RPC.\n"
+            "  ...,                                 same output as verbosity = 2\n"
```
```suggestion
            "  ...,                                 same output as verbosity = 1\n"
```

cronicc commented on src/rpc/blockchain.cpp:
```diff
@@ -1142,8 +1203,17 @@ UniValue getglobaltips(const UniValue& params, bool fHelp)
+            
             "\nExamples:\n"
             + HelpExampleCli("getglobaltips", "\"hash\"")
+            + HelpExampleRpc("getglobaltips", "\"hash\"")
```
```suggestion
            + HelpExampleCli("getglobaltips", "")
            + HelpExampleRpc("getglobaltips", "")
```
Takes no arguments.

cronicc commented on src/rpc/blockchain.cpp:
```diff
@@ -1077,8 +1130,16 @@ UniValue getblockfinalityindex(const UniValue& params, bool fHelp)
+            "1. hash   (string, required)  the block hash"
+
+            "\"Result:\n"
+            "n         (numeric) number of consecutive blocks a miner should mine from now in order to revert the block of given hash\n"
```
```suggestion
            "\nReturns the minimum number of consecutive blocks a miner would have to mine from now in order to revert the block of given hash\n"

            "\nArguments:\n"
            "1. hash   (string, required)  the block hash"

            "\"Result:\n"
            "n         (numeric) number of consecutive blocks a miner would have to mine from now in order to revert the block of given hash\n"
```

cronicc commented on src/rpc/mining.cpp:
```diff
@@ -849,18 +887,22 @@ UniValue getblocksubsidy(const UniValue& params, bool fHelp)
+            "  \"miner\": xxxx,           (numeric) the mining reward amount in " + CURRENCY_UNIT + "\n"
+            "  \"community\": xxxx,       (numeric) the community fund amount in " + CURRENCY_UNIT + "\n"
+            "  \"securenodes\": xxxx,     (numeric) the securenodes fund amount in " + CURRENCY_UNIT + " (only if securenode HF fund is available)\n"
+            "  \"supernodes\": xxxx       (numeric) the supernodes fund amount in " + CURRENCY_UNIT + " (only if supernode HF fund is available)\n"
```
```suggestion
            "  \"securenodes\": xxxx,     (numeric) the securenodes fund amount in " + CURRENCY_UNIT + "\n"
            "  \"supernodes\": xxxx       (numeric) the supernodes fund amount in " + CURRENCY_UNIT + "\n"
```
As the fork is active now this can be removed.

cronicc commented on src/rpc/misc.cpp:
```diff
@@ -151,24 +153,28 @@ UniValue validateaddress(const UniValue& params, bool fHelp)
-            "validateaddress \"zcashaddress\"\n"
-            "\nReturn information about the given Zcash address.\n"
+            "validateaddress \"zenaddress\"\n"
+            "\nVerify an ZEN address is valid.\n"
```
```suggestion
            "\nReturn information about the given ZEN address.\n"
```
Original wording was OK.

cronicc commented on src/rpc/misc.cpp:
```diff
@@ -215,21 +221,23 @@ UniValue z_validateaddress(const UniValue& params, bool fHelp)
         throw runtime_error(
             "z_validateaddress \"zaddr\"\n"
-            "\nReturn information about the given z address.\n"
+            "\nVerify an zaddress is valid.\n"
```
```suggestion
            "\nReturn information about the given zaddress.\n"
```
Original wording was OK.

cronicc commented on src/rpc/misc.cpp:
```diff
@@ -440,9 +450,17 @@ UniValue setmocktime(const UniValue& params, bool fHelp)
-            "1. timestamp  (integer, required) Unix seconds-since-epoch timestamp\n"
-            "   Pass 0 to go back to using the system time."
+            "1. timestamp  (numeric, required) Unix seconds-since-epoch timestamp\n"
+            "               pass 0 to go back to using the system tim."
```
```suggestion
            "               pass 0 to go back to using the system time."
```

cronicc commented on src/rpc/net.cpp:
```diff
@@ -81,37 +87,40 @@ UniValue getpeerinfo(const UniValue& params, bool fHelp)
+            "    \"addrlocal\": \"ip:port\",             (string) local address\n"
+            "    \"services\":\"xxxxxxxxxxxxxxxx\",      (string) the services offered\n"
+            "    \"tls_established\": true|false,        (boolean) status of TLS connection\n"
+            "    \"tls_verified\": true|false,           (boolean) status of peer certificate. Will be true if a peer certificate can be verified with some trusted root certs \n"
```
```suggestion
            "    \"tls_verified\": true|false,           (boolean) status of peer certificate. True if the chain of trust of a peer certificate can be verified using the OS certificate store\n"
```

cronicc commented on src/rpc/net.cpp:
```diff
@@ -225,8 +239,13 @@ UniValue disconnectnode(const UniValue& params, bool fHelp)
+            
             "\nExamples:\n"
             + HelpExampleCli("disconnectnode", "\"192.168.0.6:8233\"")
             + HelpExampleRpc("disconnectnode", "\"192.168.0.6:8233\"")
```
```suggestion
            + HelpExampleCli("disconnectnode", "\"192.168.0.6:9033\"")
            + HelpExampleRpc("disconnectnode", "\"192.168.0.6:9033\"")
```

cronicc commented on src/rpc/net.cpp:
```diff
@@ -408,35 +432,38 @@ UniValue getnetworkinfo(const UniValue& params, bool fHelp)
+            "  \"timeoffset\": 0,                             (numeric) the time offset (deprecated; always 0)\n"
+            "  \"connections\": xxxxx,                        (numeric) the total number of open connections for the node\n"
+            "  \"tls_cert_verified\": true|flase,             (boolean) true if the certificate of the current node is verified\n"
+            "  \"networks\": [                                (array) an array of objects describing IPV4, IPV6 and Onion network interface states\n"
```
```suggestion
            "  \"networks\": [                                (array) an array of objects describing IPV4, IPV6 and Onion network status\n"
```

cronicc commented on src/rpc/net.cpp:
```diff
@@ -480,18 +507,23 @@ UniValue setban(const UniValue& params, bool fHelp)
+            
+            "\nArguments:\n"
+            "1. \"ip(/netmask)\" (string, required) the IP/Subnet (see getpeerinfo for nodes ip) with a optional netmask (default is /32 = single ip)\n"
+            "2. \"command\"      (string, required) 'add' to add a IP/Subnet to the list, 'remove' to remove a IP/Subnet from the list\n"
```
```suggestion
            "\nAttempts to add or remove an IP/Subnet from the banned list.\n"
            
            "\nArguments:\n"
            "1. \"ip(/netmask)\" (string, required) the IP/Subnet (see getpeerinfo for nodes ip) with an optional netmask (default is /32 = single ip)\n"
            "2. \"command\"      (string, required) 'add' to add an IP/Subnet to the list, 'remove' to remove an IP/Subnet from the list\n"
```

cronicc commented on src/rpc/rawtransaction.cpp:
```diff
@@ -180,78 +180,78 @@ UniValue getrawtransaction(const UniValue& params, bool fHelp)
-            "         \"asm\": \"asm\",  (string) asm\n"
-            "         \"hex\": \"hex\"   (string) hex\n"
+            "       \"txid\": \"id\",             (string) the transaction id\n"
+            "       \"vout\": n,                  (numeric) the output number\n"
```
```suggestion
            "       \"vout\": n,                  (numeric) the output index\n"
```

cronicc commented on src/rpc/rawtransaction.cpp:
```diff
@@ -180,78 +180,78 @@ UniValue getrawtransaction(const UniValue& params, bool fHelp)
             "     ,...\n"
             "  ],\n"
-            "  \"vjoinsplit\" : [        (array of json objects, only for version >= 2)\n"
+            "  \"vjoinsplit\" : [                 (array of json objects, only for version >= 2)\n"
```
```suggestion
            "  \"vjoinsplit\" : [                 (array of json objects, only for version 2 or -3)\n"
```

cronicc commented on src/rpc/rawtransaction.cpp:
```diff
@@ -484,64 +496,64 @@ UniValue decoderawtransaction(const UniValue& params, bool fHelp)
-            "         \"asm\": \"asm\",  (string) asm\n"
-            "         \"hex\": \"hex\"   (string) hex\n"
+            "       \"txid\": \"id\",                             (string) the transaction id\n"
+            "       \"vout\": n,                                  (numeric) the output number\n"
```
```suggestion
            "       \"vout\": n,                                  (numeric) the output index\n"
```

cronicc commented on src/rpc/rawtransaction.cpp:
```diff
@@ -484,64 +496,64 @@ UniValue decoderawtransaction(const UniValue& params, bool fHelp)
             "     ,...\n"
             "  ],\n"
-            "  \"vjoinsplit\" : [        (array of json objects, only for version >= 2)\n"
+            "  \"vjoinsplit\": [                                  (array of json objects, only for version >= 2)\n"
```
```suggestion
            "  \"vjoinsplit\": [                                  (array of json objects, only for version 2 or -3)\n"
```

cronicc commented on src/rpc/rawtransaction.cpp:
```diff
@@ -637,23 +651,23 @@ UniValue signrawtransaction(const UniValue& params, bool fHelp)
-            "2. \"prevtxs\"       (string, optional) An json array of previous dependent transaction outputs\n"
-            "     [               (json array of json objects, or 'null' if none provided)\n"
+            "1. \"hexstring\"                      (string, required) the transaction hex string\n"
+            "2. \"prevtxs\"                        (string, optional) an json array of previous dependent transaction outputs\n"
```
```suggestion
            "2. \"prevtxs\"                        (string, optional) a json array of previous dependent transaction outputs\n"
```

cronicc commented on src/wallet/rpcdump.cpp:
```diff
@@ -746,15 +782,18 @@ UniValue z_exportkey(const UniValue& params, bool fHelp)
             + HelpExampleCli("z_exportkey", "\"myaddress\"")
-            + HelpExampleCli("z_importkey", "\"mykey\"")
-            + HelpExampleRpc("z_exportkey", "\"myaddress\"")
+            + HelpExampleCli("z_importkey", "\"zaddr\"")
```
```suggestion
            + HelpExampleCli("z_importkey", "\"zkey\"")
```

cronicc commented on src/wallet/rpcwallet.cpp:
```diff
@@ -266,9 +274,14 @@ UniValue setaccount(const UniValue& params, bool fHelp)
+            
             "\nExamples:\n"
             + HelpExampleCli("setaccount", "\"t14oHp2v54vfmdgQ3v3SNuQga8JKHTNi2a1\" \"tabby\"")
             + HelpExampleRpc("setaccount", "\"t14oHp2v54vfmdgQ3v3SNuQga8JKHTNi2a1\", \"tabby\"")
```
```suggestion
            + HelpExampleCli("setaccount", "\"horizenaddress\" \"tabby\"")
            + HelpExampleRpc("setaccount", "\"horizenaddress\", \"tabby\"")
```

luisantoniocrag committed:
Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/misc.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/misc.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/misc.cpp

typo

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/wallet/rpcdump.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/mining.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/wallet/rpcwallet.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Update RPC ports

Fix /rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Fix src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

Fix some typos and small changes

luisantoniocrag committed:
Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/misc.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/misc.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/misc.cpp

typo

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/wallet/rpcdump.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/mining.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/wallet/rpcwallet.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update src/rpc/rawtransaction.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Update RPC ports

cronicc commented on src/rpc/blockchain.cpp:
```diff
@@ -1076,9 +1129,17 @@ UniValue getblockfinalityindex(const UniValue& params, bool fHelp)
+            "\nArguments:\n"
+            "1. hash   (string, required)  the block hash"
+
+            "\"Result:\n"
```
```suggestion
            "1. hash   (string, required)  the block hash\n"

            "\nResult:\n"
```

cronicc commented on src/rpc/net.cpp:
```diff
@@ -81,37 +87,40 @@ UniValue getpeerinfo(const UniValue& params, bool fHelp)
             "getpeerinfo\n"
             "\nReturns data about each connected network node as a json array of objects.\n"
+            
             "\nbResult:\n"
```
```suggestion
            "\nResult:\n"
```

cronicc reviewed (changes requested):
@luisantoniocrag 
Hi, I've found two more small changes to make.

Once these are addressed, please rebase (squash) all of the commits from the change requests down to a single commit.

Here is a useful tutorial on squashing http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

luisantoniocrag committed:
Fix /rpc/blockchain.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Fix src/rpc/net.cpp

Co-authored-by: cronicc <cronic@zensystem.io>

luisantoniocrag committed:
Merge branch 'review_rpc_call_msgs' of https://github.com/luisantoniocrag/zen into review_rpc_call_msgs
